dev/io.openliberty.webcontainer.servlet.6.0.internal_fat/fat/src/io/openliberty/webcontainer/servlet60/fat/tests/Servlet60CookieSetAttributeTest.java

```diff
@@ -0,0 +1,179 @@
+        LOG.info("\n Sending Request [" + url + "]");
+        HttpGet getMethod = new HttpGet(url);
+
+        try (CloseableHttpClient client = HttpClientBuilder.create().build()) {
+            try (CloseableHttpResponse response = client.execute(getMethod)) {
+                String responseText = EntityUtils.toString(response.getEntity());
+                LOG.info("\n" + "Response Text: [" + responseText + "]");
+
+                //fail-fast check
+                assertTrue("The response did not contain the following String: " + expectedGeneralResponse, responseText.contains(expectedGeneralResponse));
+
+                Header[] headers = response.getHeaders();
+                for (Header header : headers) {
+                    LOG.info("\n" + "Header: " + header);
+                }
+
+                String headerValue = response.getHeader("Set-Cookie").getValue();
+
+                LOG.info("\n Set-Cookie value [" + headerValue + "]");
+
+                assertTrue("The response did not contain the following Set-Cookie header " + expectedSetCookie, headerValue.contains(expectedSetCookie));
+
+            }
+        }
+    }
+
+    /*
+     * Test Set-Cookie via addHeader and setHeader behavior do not change in Servlet 6.0
+     */
+    @Test
+    public void test_cookieAddAndSetHeader() throws Exception {
+        String url = "http://" + server.getHostname() + ":" + server.getHttpDefaultPort() + "/" + TEST_APP_NAME + "/CookieSetAttributeServlet?testName=addAndSetHeaders";
+        LOG.info("\n Sending Request [" + url + "]");
+        HttpGet getMethod = new HttpGet(url);
+
+        try (CloseableHttpClient client = HttpClientBuilder.create().build()) {
+            try (CloseableHttpResponse response = client.execute(getMethod)) {
+                String responseText = EntityUtils.toString(response.getEntity());
+                LOG.info("\n" + "Response Text: [" + responseText + "]");
+
+                //fail-fast check
+                assertTrue("The response did not contain the following String: " + expectedGeneralResponse, responseText.contains(expectedGeneralResponse));
+
+                Header[] headers = response.getHeaders();
+                String headerValue;
+                int hitFound = 0;
+
+                for (Header header : headers) {
+                    LOG.info("\n" + "Header: " + header);
+                    if (header.getName().equals("Set-Cookie")) {
+                        headerValue = header.getValue();
+                        LOG.info("\n" + "Set-Cookie value: " + headerValue);
+                        if (headerValue.equals("randomAttributeA=myAttValueA") || headerValue.equals("randomAttributeB=myAttValueB")) {
+                            hitFound++;
+                        }
+                    }
+                }
+
+                assertTrue("Expecting 2 Set-Cookie headers for random attribute , but found [" + hitFound + "]", hitFound == 2);
```

Comment:
Phu, can you get an issue opened for us to look at potentially changing this for Servlet 6.0 +?

Reply:
Opened https://github.com/OpenLiberty/open-liberty/issues/22324